internal/pkg/template/workload.go

```diff
@@ -483,6 +489,114 @@ func (cfg *ALBListener) RulePaths() []string {
 	return rulePaths
 }
 
+// ConditionGroups returns a slice of all the Splits on the Conditions across multiple listener rules.
+func (cfg *ALBListener) ConditionGroups(wkldType string, isHTTPS bool) []string {
+	var groupCount []string
+	for _, rule := range cfg.Rules {
+		groupCount = append(groupCount, strconv.Itoa(len(rule.GenerateConditionGroups(wkldType, isHTTPS))))
+	}
+	return groupCount
+}
+
+// conditionGroup represents groups of conditions per listener rule.
+type conditionGroup struct {
+	AllowedSourceIps []string
+	Aliases          []string
+}
+
+// GenerateConditionGroups will split allowed source ips and aliases into separate condition groups per listener rule.
+// This is needed because Listener rule has quota of 5 conditions per rule.
+// https://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-limits.html
+func (lr ALBListenerRule) GenerateConditionGroups(wkldType string, isHTTPS bool) []conditionGroup {
+	var groups []conditionGroup
+	remaining := lr.calculateRemainingConditions(wkldType, isHTTPS)
+	if len(lr.Aliases)+len(lr.AllowedSourceIps) <= remaining {
```

Comment:
I _think_ there is no relation between the number of aliases and the number of allowed source IPs. i.e. you can have:
```yaml
Conditions:
      - Field: 'source-ip'
        SourceIpConfig:
          Values: [ip1, ip2, ip3, ip4, ip5]
      - Field: 'host-header'
        HostHeaderConfig:
          Values: [alias1, alias2, alias3, alias4, alias5]
```